### PR TITLE
Fix proxy scroll

### PR DESCRIPTION
**Pull request by markbrocato**
See https://moovweb.atlassian.net/browse/PRPL-2003

This fixes the issue where the page would scroll to the top when a link points to a route with a proxyUpstream handler.  The page should just reload but the scroll position should stay the same.

### CODE REVIEW

File: packages/react-storefront/src/router/Router.js [hunk 1]
```diff
@@ -448,18 +458,6 @@ export default class Router extends EventEmitter {
     this.emit('after', { request, response: context })
   }
 
-  /**
-   * Call this function when client-side navigation to a non-PWA route has occurred.  This
-   * will revert window.history to the previous url and reload the original destination URL
-   * from the server.
-   * @private
-   */
-  reloadFromServer() {
```
Comment (dijs) [inline packages/react-storefront/src/router/Router.js]: Why was this removed?

Comment (markbrocato) [reply packages/react-storefront/src/router/Router.js]: It turns out all I needed to do was location.reload() and the only place that needed to do it was proxyUpstream on the client, so I felt it was more obvious to do it there.
